src/crimson/tools/store_bench/store-bench.cc

```diff
@@ -0,0 +1,685 @@
+      auto coll_id = make_cid(i);
+      collection_id[i] = coll_id;
+      auto coll_ref = co_await local_store.create_new_collection(coll_id);
+      coll_ref_map[i] = coll_ref;
+      std::map<std::string, bufferlist> data;
+      for (int j = 0; j < log_length; ++j) {
+        std::string key = std::to_string(j);
+        bufferlist bl_value;
+        bl_value.append_zero(log_size);
+        data[key] = bl_value;
+      }
+      ceph::os::Transaction txn;
+      txn.create(coll_id, obj_i);
+      txn.omap_setkeys(coll_id, obj_i, data);
+      co_await local_store.do_transaction(coll_ref, std::move(txn));
+    }
+    co_return;
+  };
+
+  std::vector<int> first_key_per_log(num_logs,
+                                     0); // first key in each log object
+  std::vector<int> last_key_per_log(num_logs,
+                                    log_length); // last key in each log object
+
+  /**
+   * This method returns a future of type struct results_t
+   * In this function we choose a random log object to write to and remove keys
+   * from We add and remove keys sequentially, add keys to the end, remove from
+   * the front Total latency per io is calculated as the sum of the time it
+   * takes to add and remove each key
+   */
+  auto add_remove_entry = [&]() -> seastar::future<results_t> {
+    int num_ops = 0;
+    int tot_latency = 0;
+    auto start = seastar::lowres_clock::now();
+
+    while (seastar::lowres_clock::now() - start <=
+           std::chrono::milliseconds(duration)) {
+      int obj_num = std::rand() % num_logs;
+
+      auto object = create_hobj(obj_num);
+      auto coll_id = collection_id[obj_num];
+      auto coll_ref = coll_ref_map[obj_num];
+
+      std::string key_to_write = std::to_string(last_key_per_log[obj_num]);
+      last_key_per_log[obj_num] += 1;
+
+      std::string key_to_remove = std::to_string(first_key_per_log[obj_num]);
+      first_key_per_log[obj_num] += 1;
+
+      bufferlist val;
+      val.append_zero(log_size);
+      std::map<std::string, bufferlist> key_val;
+      key_val[key_to_write] = val;
+
+      ceph::os::Transaction one_write_delete;
+      one_write_delete.omap_setkeys(coll_id, object, key_val);
+      one_write_delete.omap_rmkey(coll_id, object, key_to_remove);
+
+      auto latency_start = seastar::lowres_clock::now();
+      co_await local_store.do_transaction(coll_ref,
+                                          std::move(one_write_delete));
+      auto latency_end = seastar::lowres_clock::now();
+
+      auto time_millisec =
+          std::chrono::duration_cast<std::chrono::milliseconds>(latency_end -
+                                                                latency_start)
+              .count();
+      // ERROR("shreya :completed 1 IO");
+      tot_latency += time_millisec;
+      num_ops++;
+    }
+    co_return results_t{num_ops, tot_latency};
+  };
+  co_await pre_fill_logs();
+  co_await run_concurrent_ios(duration, num_concurrent_io, add_remove_entry);
```

Comment:
With this version, run_concurrent_ios prints the results to stderr using ERROR.  Instead, it would be more convenient for automation if the utility printed a JSON representation of the result to stdout.  See common/Formatter.h JSONFormatter.